source/parser/x86_64/classifiers.hpp

```diff
@@ -0,0 +1,105 @@
+// Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+// Spack Project Developers. See the top-level COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+#pragma once
+
+#include <utility>
+
+#include "Type.h"
+#include "register_class.hpp"
+
+namespace smeagle::x86_64 {
+
+  struct Class {
+    RegisterClass lo, hi;
```

Comment:
Could there ever be cases in the future of having more than two registers that would warrant this approach a not great idea?

Reply:
The rules for classifying aggregates (at least on x86) require looking at only two eightbytes at a time, so there would only ever be at most two.